iac-extensions/cloudformation/src/test/java/org/sonar/iac/cloudformation/plugin/CfnLintImporterTest.java

```diff
@@ -54,28 +61,42 @@ void setUp() throws IOException {
 
   @ParameterizedTest
   @CsvSource({
-    "src/test/resources/cfn-lint/doesNotExist.json, Cfn-lint report importing: path does not seem to point to a file %s",
-    "src/test/resources/cfn-lint/parseError.json, Cfn-lint report importing: could not parse file as JSON %s",
-    "src/test/resources/cfn-lint/noArray.json, Cfn-lint report importing: file is expected to contain a JSON array but didn't %s",
-  })
-  void problem_when_reading_or_parsing_file(String reportPath, String expectedLog) {
-    File reportFile = new File(reportPath);
-    CfnLintImporter.importReport(context, reportFile);
+    "src/test/resources/cfn-lint/doesNotExist.json, src\\test\\resources\\cfn-lint\\doesNotExist.json, Cfn-lint report importing: path does not seem to point to a file %s",
+    "src/test/resources/cfn-lint/parseError.json, src\\test\\resources\\cfn-lint\\parseError.json, Cfn-lint report importing: could not parse file as JSON %s",
+    "src/test/resources/cfn-lint/noArray.json, src\\test\\resources\\cfn-lint\\noArray.json, Cfn-lint report importing: file is expected to contain a JSON array but didn't %s"})
+  void problem_when_reading_or_parsing_file(String reportPath, String reportPathWindows, String expectedLog) {
+    String path = File.separatorChar == '/' ? reportPath : reportPathWindows;
+    File reportFile = new File(path);
+
+    CfnLintImporter.importReport(context, reportFile, spyAnalysisWarnings);
+    assertThat(logTester.logs(LoggerLevel.WARN))
+      .containsExactly(String.format(expectedLog, path));
+  }
+
+  @Test
+  void reading_issue(){
+    String path = "src\\test\\resources\\cfn-lint\\throwsIOException.json";
+    File reportFile = Mockito.mock(File.class);
+    when(reportFile.getPath()).thenReturn(path);
+    when(reportFile.isFile()).thenReturn(true);
+    doAnswer((invocation) -> {throw new IOException();}).when(reportFile).toPath();
```

Comment:
When I used the function you mentioned, I ran into this exception : `org.mockito.exceptions.base.MockitoException: 
Checked exception is invalid for this method!`.
The entry point `reportFile.toPath()` must be used as it is the only one we can easily trigger so that the wanted exception is raised. The problem is that `toPath()` is not expected to throw an IOException, resulting in this invalid message (either `JSONParser.parse()` or `Files.newBufferedReader()` is expected to throw this exception, but we don't have control over them). I had to get around this by using `doAnswer()` which allows to perform some actions when `toPath()` is called, such as throwing an "unrelated" exception.